### ISSUE TITLE
Provide a way to add multiple `QueryParams` when using `WebClient.prepare()`

### ISSUE THREAD

**Issue by jrhee17** [line/armeria#4160]
Some users may want to add multiple query param values for a single key

e.g. `localhost:8080/v1?id=1&id=2&id=3`

However, `WebClient.prepare().queryParam()` performs a `QueryParamsBuilder#setObject` by default, making it
difficult to execute such a query programmatically

```
WebClient.prepare()
         .get("/")
         .queryParam("id", 1)
         .queryParam("id", 2)
         .queryParam("id", 3)
         .execute();
```

Wondering if we want to modify to use `QueryParamsBuilder#addObject` under the hood.

**Comment by ikhoon** [comment]
I believe many people won't expect the chained `queryParam()`s override the previous one.
Furthermore, `.header("X-Foo-Header", 1).header("X-Foo-Header", 2)` combines two headers into one.
https://github.com/line/armeria/blob/5c521cc47cf57ac17d22a2b87ffccf6ba036ee57/core/src/main/java/com/linecorp/armeria/common/AbstractHttpMessageBuilder.java#L63-L64
So I prefer to change the behavior to `addObject` from `setObject` for consistency and usability.

**Comment by doppany** [comment]
@ikhoon @jrhee17 
Hi! I'd like to try this issue. May I take it?

**Comment by ikhoon** [comment]
Sorry. Seems like @Jsing is going to send a PR for this issue.

**Comment by doppany** [comment]
Okay. I will bide time until there is another issue interesting to me 🙇

**Comment by Jsing** [comment]
Oh I'm sorry. I didn't mean to do PR. I'm new to contributing to an open source project. I just pushed it to my repository and didn't know it was showing up in this repository. it's yours!

**Comment by Jsing** [comment]
I am interested in the Armeria project, so I am just studying through this issue. I am so sorry.

**Comment by ikhoon** [comment]
Don't worry. I believe @doppany will understand the situation. 🙇‍♂️
I think you have already finished your work. Would you mind making your change a PR?

**Comment by doppany** [comment]
@Jsing I think it would be better if you keep going for this issue 😆. If you create the first PR, I will be happy too! I'll look into other issues 🧐 , don't worry.

**Comment by Jsing** [comment]
Thanks @doppany @ikhoon @jrhee17 😊

I have made my first PR. Everything is a bit unfamilliar and I'm not sure if I followed the guide well. I am ready to receive any advice sweetly.  #4173

**Comment by ikhoon** [comment]
Fixed by #4173
